### PR TITLE
Do not catch NoSuchElementException in HgOpenJDKImport

### PR DESCRIPTION
**Pull request by edvbld**
Hi all,

this patch removes catching `NoSuchElementException` in the `main` method for `HgOpenJDKImport` to ease debugging.

## Testing
- [x] `sh gradlew test` passes on Linux x86_64

Thanks,
Erik

Progress
--------
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

Approvers
---------
 * Robin Westberg ([rwestberg](@rwestberg) - **Reviewer**)

### CODE REVIEW

File: cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java [hunk 1]
```diff
@@ -36,16 +36,11 @@
 import java.util.logging.*;
 
 public class HgOpenJDKImport {
-    private static void die(Exception e) {
-        System.err.println(e.getMessage());
-        System.exit(1);
-    }
-
     private static Supplier<NoSuchElementException> error(String fmt, Object... args) {
```
Comment (rwestberg) [inline cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java]: Pre-existing, but it's a bit unclear why NoSuchElementException is used - plain RuntimeException (or Exception) would probably make more sense? Or perhaps something like NoSuchFileException?

File: cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java [hunk 2]
```diff
@@ -36,16 +36,11 @@
...
         return () -> new NoSuchElementException(String.format(fmt, args));
     }
 
-    public static void main(String[] args) throws IOException {
+    public static void main(String[] args) throws IOException, NoSuchElementException {
```
Comment (rwestberg) [inline cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java]: And if you go with a RuntimeException-based one I guess there's no need to declare it here either.

File: cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java (hunk 1 above)
Comment (edvbld) [reply cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java]: Yes, that is a bit unclear to me as well, and I even wrote the code in the first place :laughing: I _think_ I wanted my `catch` clause to catch `NoSuchElementException` so I could just do `.orElseThrow()` or `.get()` and still display a somewhat nice error message automatically. Pretty bad idea in retrospect, I will rework this.

File: cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java (hunk 2 above)
Comment (edvbld) [reply cli/src/main/java/org/openjdk/skara/cli/HgOpenJDKImport.java]: Yes, you are totally right, thanks!

**Pull request closed without merging**